integrationTests/otpRateLimiting_test.go

```diff
@@ -315,3 +345,73 @@ func TestOTPRateLimiting_TimeControl(t *testing.T) {
 		require.Equal(t, expOtpVerifyData, otpVerifyData)
 	})
 }
+
+func TestMultipleInstanceConcurrency(t *testing.T) {
+	t.Parallel()
+
+	maxFailures := 3
+	periodLimit := 9
+
+	server := miniredis.RunT(t)
+	redisClient := redis.NewClient(&redis.Options{
+		Addr: server.Addr(),
+	})
+	redisLimiter1, err := redisLocal.NewRedisClientWrapper(redisClient)
+	require.Nil(t, err)
+
+	rateLimiterArgs1 := redisLocal.ArgsRateLimiter{
+		OperationTimeoutInSec: 1000,
+		MaxFailures:           int64(maxFailures),
+		LimitPeriodInSec:      uint64(periodLimit),
+		Storer:                redisLimiter1,
+	}
+	rl1, err := redisLocal.NewRateLimiter(rateLimiterArgs1)
+	require.Nil(t, err)
+
+	redisLimiter2, err := redisLocal.NewRedisClientWrapper(redisClient)
+	require.Nil(t, err)
+
+	rateLimiterArgs2 := redisLocal.ArgsRateLimiter{
+		OperationTimeoutInSec: 1000,
+		MaxFailures:           int64(maxFailures),
+		LimitPeriodInSec:      uint64(periodLimit),
+		Storer:                redisLimiter2,
+	}
+	rl2, err := redisLocal.NewRateLimiter(rateLimiterArgs2)
+	require.Nil(t, err)
+
+	numOps := 50000
+	wg := sync.WaitGroup{}
+
+	wg.Add(numOps)
+
+	cnt := uint32(0)
+	key := "key1"
+	for i := 0; i < numOps; i++ {
+		go func(idx int) {
+			switch idx % 6 {
+			case 0, 1:
+				_, err := rl1.CheckAllowedAndIncreaseTrials(key)
+				if err == redisLocal.ErrNoExpirationTimeForKey {
+					atomic.AddUint32(&cnt, 1)
+				}
+			case 2, 3:
```

Comment:
leave only one value per case here?

Reply:
I tried to unbalance the calls, not to have one reset for each check, but rather more checks than resets